workloads/kube-burner/common.sh

```diff
@@ -0,0 +1,133 @@
+  log "Deploying benchmark"
+  envsubst < kube-burner-crd.yaml | oc apply -f -
+}
+
+wait_for_benchmark() {
+  rc=0
+  log "Waiting for kube-burner job to be created"
+  until oc get benchmark -n my-ripsaw kube-burner-${1}-${UUID} -o jsonpath="{.status.state}" | grep -q Running; do
+    sleep 1
+  done
+  log "Waiting for kube-burner job to start"
+  suuid=$(oc get benchmark -n my-ripsaw kube-burner-${1}-${UUID} -o jsonpath="{.status.suuid}")
+  until oc get pod -n my-ripsaw -l job-name=kube-burner-${suuid} --ignore-not-found -o jsonpath="{.items[*].status.phase}" | grep -q Running; do
+    sleep 1
+  done
+  if [[ ${LOG_STREAMING} == "true" ]]; then
+    oc logs -n my-ripsaw -f -l job-name=kube-burner-${suuid}
+  fi
+  log "Benchmark in progress, Waiting for benchmark/kube-burner-${1}-${UUID} object to be updated"
+  until oc get benchmark -n my-ripsaw kube-burner-${1}-${UUID} -o jsonpath="{.status.state}" | grep -Eq "Complete|Failed"; do
+    sleep 1
+  done
+  if [[ ${LOG_STREAMING} == "false" ]]; then
+    oc logs -n my-ripsaw --tail=-1 -l job-name=kube-burner-${suuid}
+  fi
+  oc get pod -l job-name=kube-burner-${suuid} -n my-ripsaw
+  status=$(oc get benchmark -n my-ripsaw kube-burner-${1}-${UUID} -o jsonpath="{.status.state}")
+  log "Benchmark kube-burner-${1}-${UUID} finished with status: ${status}"
+  if [[ ${status} == "Failed" ]]; then
+    rc=1
+  fi
+}
+
+label_nodes() {
+  export NODE_SELECTOR_KEY="kubelet-density"
+  export NODE_SELECTOR_VALUE="enabled"
+  if [[ ${NODE_COUNT} -le 0 ]]; then
+    log "Node count <= 0: ${NODE_COUNT}"
+    exit 1
+  fi
+  nodes=$(oc get node -o name -l node-role.kubernetes.io/worker= | head -${NODE_COUNT})
+  if [[ $(echo "${nodes}" | wc -l) -lt ${NODE_COUNT} ]]; then
+    log "Not enough worker nodes to label"
+    exit 1
+  fi
+  for n in ${nodes}; do
+    pods=$(oc describe ${n} | awk '/Non-terminated/{print $3}' | sed "s/(//g")
+    pod_count=$((pods + pod_count))
+  done
+  total_pod_count=$((PODS_PER_NODE * NODE_COUNT - pod_count))
+  log "Total running pods across nodes: ${pod_count}"
+  if [[ ${total_pod_count} -le 0 ]]; then
+    log "Number of pods to deploy <= 0"
+    exit 1
+  fi
+  log "Number of pods to deploy on nodes: ${total_pod_count}"
+  if [[ ${1} == "heavy" ]]; then
+    total_pod_count=$((total_pod_count / 2))
+  fi
+  export JOB_ITERATIONS=${total_pod_count}
+  log "Labeling ${NODE_COUNT} worker nodes with kubelet-density=enabled"
+  for n in ${nodes}; do
+    oc label ${n} kubelet-density=enabled --overwrite
+  done
+}
+
+unlabel_nodes() {
+  log "Removing kubelet-density=enabled label from worker nodes"
+  for n in ${nodes}; do
+    oc label ${n} kubelet-density-
+  done
+}
+
+check_running_benchmarks() {
+  benchmarks=$(oc get benchmark -n my-ripsaw --ignore-not-found | wc -l)
+  if [[ ${benchmarks} -gt 0 ]]; then
+    echo ${benchmarks} | grep kube-burner | grep -qE "Failed|Completed" || log "Another kube-burner benchmark is running at the moment" && exit 1
```

Comment:
Think we need to grep for benchmarks in "Running" state?

Reply:
There's a period of time a benchmark doesn't have state. Grep-ing for Failed and Complete is the best way to be sure there's no benchmark running at the moment